crates/context/config/src/client/env/config/types/starknet.rs

```diff
@@ -0,0 +1,484 @@
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationId {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+// Similar implementations for ApplicationBlob
+impl From<crate::types::BlobId> for ApplicationBlob {
+    fn from(value: crate::types::BlobId) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationBlob {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+impl From<Repr<crate::types::BlobId>> for ApplicationBlob {
+    fn from(value: Repr<crate::types::BlobId>) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationBlob {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+impl From<&Repr<crate::types::BlobId>> for ApplicationBlob {
+    fn from(value: &Repr<crate::types::BlobId>) -> Self {
+        let bytes = value.as_bytes();
+        let (high_bytes, low_bytes) = bytes.split_at(bytes.len() / 2);
+        ApplicationBlob {
+            high: Felt::from_bytes_be_slice(high_bytes),
+            low: Felt::from_bytes_be_slice(low_bytes),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct EncodableString(pub String);
+
+impl From<crate::types::ApplicationSource<'_>> for EncodableString {
+    fn from(value: crate::types::ApplicationSource<'_>) -> Self {
+        EncodableString(value.0.into_owned())
+    }
+}
+
+impl From<crate::types::ApplicationMetadata<'_>> for EncodableString {
+    fn from(value: crate::types::ApplicationMetadata<'_>) -> Self {
+        EncodableString(String::from_utf8_lossy(&value.0.into_inner()).into_owned())
+    }
+}
+
+// Optional: Add this if you need to convert from string references
+impl From<&str> for EncodableString {
+    fn from(value: &str) -> Self {
+        EncodableString(value.to_string())
+    }
+}
+
+impl Encode for EncodableString {
+    fn encode<W: FeltWriter>(&self, writer: &mut W) -> Result<(), Error> {
+        const WORD_SIZE: usize = 31;
+        let bytes = self.0.as_bytes();
+
+        // Calculate full words and pending word
+        let full_words_count = bytes.len() / WORD_SIZE;
+        let pending_len = bytes.len() % WORD_SIZE;
+
+        // Write number of full words
+        writer.write(Felt::from(full_words_count));
+
+        // Write full words (31 chars each)
+        for i in 0..full_words_count {
+            let start = i * WORD_SIZE;
+            let word_bytes = &bytes[start..start + WORD_SIZE];
+            let word_hex = hex::encode(word_bytes);
+            writer.write(Felt::from_hex(&format!("0x{}", word_hex)).unwrap());
```

Comment:
why `bytes -> hex -> felt` when we can `bytes -> felt`? is it because it's 31 bytes? does it work if we prepend or append a zeroed 32nd byte?

Reply:
yes, the thing was that if we have 32bytes it will overflow, same like we had for other variables that needed to be split into two parts. Since there is no decode function out-of-the-box for ByteArray/String type I needed to introduce this.